src/components/PopupWithForm.js

```diff
@@ -7,44 +7,42 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._inputs = this._popupElement.querySelectorAll(".modal__input");
     this._submitButton = this._popupElement.querySelector(
-      ".modal__submit-button"
+      ".modal__button"
     );
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   close() {
-    this._popupForm.reset();
     super.close();
+    this._popupForm.reset();
   }
 
   _getInputValues() {
     const _inputs = Array.from(
-      this.popupForm.querySelectorAll(".modal__input")
+      this._popupForm.querySelectorAll(".modal__input")
     );
     this._inputValues = {};
       _inputs.forEach((input) => {
-      this._inputs[input.name] = input.value;
+      this._inputValues[input.name] = input.value;
     });
-    return inputValues;
+    return this._inputValues;
   }
 
   setLoading(submit, loadingText = "Saving...") {

```

Comment:
Great that you've implemented the logic for rendering loading via the class methods. That's the best way!